ipa-core/src/protocol/ipa_prf/shuffle/base.rs

```diff
@@ -18,7 +18,10 @@ use crate::{
 
 /// # Errors
 /// Will propagate errors from transport and a few typecasts
-pub async fn shuffle<C, I, S>(ctx: C, shares: I) -> Result<Vec<AdditiveShare<S>>, Error>
+pub async fn shuffle<C, I, S>(
```

Comment:
is this function going to be the entry point for shuffle protocol or you plan to create another wrapper for it?

Reply:
For semi honest shuffle, you can call it directly (and ignore the IntermediateMessages output). For malicious shuffle, there will be a malicious shuffle function that will call this function.

Reply:
it would be better imo to avoid returning data that is not used in semi-honest setting. Maybe we could define a trait that specifies the output type for semi-honest contexts to be `Vec<AdditiveShare<S>>` and Vec<> + Intermediates for malicious

Alternatively we could explore an option to validate intermediates inside `shuffle` - for semi-honest, validation will be a no-op and for malicious it should check MACs. Then callers would just specify which protocol they want to run by passing the right context version (as with other protocols)